Restructure README.md

kbarnold opened the pull request.

kbarnold committed:
Restructure README.md to common SAP Org format

kbarnold committed:
Remove trailing whitespaces after link definitions

kbarnold committed:
Wrap lines after 80 characters

Shorter lines allow for cleaner diffs upon changes.

kbarnold committed:
Fix typo

o-liver commented on README.md:
```diff
@@ -1,69 +1,105 @@
 
-This repository contains Jenkins pipeline instructions in form of a `Jenkinsfile` (together with some auxiliary files) which require the [shared library][piper-library] of project "Piper" to run. They are available under `pipelines/<scenario>/Jenkinsfile` in this repository.
+This repository contains Jenkins pipeline instructions in form of a
+`Jenkinsfile` (together with some auxiliary files) which require the [shared
```
Comma before 'which'
My suggestion: 
"`Jenkinsfile`, together with some auxiliary files, which require the..."

kbarnold replied:
I don't want to change this since other wise, the following 'which' might be misunderstood as referencing to the auxiliary files, and not to the Jenkinsfile.

o-liver commented on README.md:
```diff
@@ -1,69 +1,105 @@
+
+There are two recommended ways to consume our `Jenkinsfile`. For both ways, you
+need to create a new pipeline job in Jenkins that [retrieves a pipeline from an
+SCM][jenkins-doc-pipelineFromSCM].  The location of the `Jenkinsfile` may
```
2 spaces before "The location..."

kbarnold replied:
Done

o-liver commented on README.md:
```diff
@@ -1,69 +1,105 @@
+   to retrieve the application source code repository.
 
-2. The `Jenkinsfile` can be kept in an own repository separate from the application. In case you want to apply the `Jenkinsfile` to multiple applications, this approach is beneficial since you only need to maintain the `Jenkinsfile` in one place. You must define two mandatory Jenkins pipeline job parameters in this case:
+2. The `Jenkinsfile` can be kept in an own repository separate from the
```
My suggestion:
"The `Jenkinsfile` can be kept in its own repository, separate..."

kbarnold replied:
'its' -> Done

I see no reason for that comma, though.

o-liver commented on README.md:
```diff
@@ -1,69 +1,105 @@
+2. The `Jenkinsfile` can be kept in an own repository separate from the
+   application. In case you want to apply the `Jenkinsfile` to multiple
+   applications, this approach is beneficial since you only need to maintain the
+   `Jenkinsfile` in one place. You can consume the `Jenkinsfile` e.g. directly from
```
Here "for example" would fit better than e.g. I believe e.g. is not equivalent to 'for example'.

kbarnold replied:
AFAIK they are equivalent.

o-liver replied:
You are right, they are.

o-liver commented on README.md:
```diff
@@ -1,69 +1,105 @@
+   applications, this approach is beneficial since you only need to maintain the
+   `Jenkinsfile` in one place. You can consume the `Jenkinsfile` e.g. directly from
+   this repository on GitHub.com or from your own fork of it. You must define two
+   mandatory Jenkins pipeline job parameters to define the location of your
```
"define" twice in this sentence. Synonyms for defined: specify, delineate, delimit, delimitate, set, determine, fix, limit, be, characterise, characterize
I vote changing the second "define" to "specify".

kbarnold replied:
Done

o-liver commented on README.md:
```diff
@@ -1,69 +1,105 @@
+   mandatory Jenkins pipeline job parameters to define the location of your
+   application:
 
     * `GIT_URL`: The URL to the Git repository in which the application resides.
```
comma before "in which" or 
"`GIT_URL`: The URL to the Git repository that the application resides in."
I like the latter more as it fits to the next point:
>`GIT_BRANCH`: The branch of the Git repository that should be checked out

kbarnold replied:
Done

o-liver commented on README.md:
```diff
@@ -1,69 +1,105 @@
-Prerequisites:
-
-* [Installed shared library][piper-library-installation] of project "Piper".
+* [Installed shared library][piper-library-readme] of project "Piper".
```
How about linking to the installation section of the readme here.
https://github.com/SAP/jenkins-library/blob/master/README.md#download-and-installation

kbarnold replied:
Done

o-liver commented on README.md:
```diff
@@ -1,69 +1,105 @@
+# Download and Installation
+
+There are two recommended ways to consume our `Jenkinsfile`. For both ways, you
+need to create a new pipeline job in Jenkins that [retrieves a pipeline from an
```
IMO it would be helpful with a screenshot of the Jenkins settings here, showing how to retrieve a pipeline from an SCM. The description of jenkins.io is not as helpful as a screenshot would be.

kbarnold replied:
Done

o-liver commented on README.md:
```diff
@@ -1,69 +1,105 @@
+    * `GIT_BRANCH`: The branch of the Git repository that should be checked out
+      for the build.
+
+# Configuration
```
Maybe a list of a few such files with a link to the file?

kbarnold replied:
A few Jenkinsfiles you mean? There is a list in the subsequent subsection of this section.

o-liver replied:
No. I was thinking of links to the template configuration files, such as 'Gruntfile', 'mta.yaml', 'package.json' etc.
Although I admit it's written in the text:
> Templates for these configuration files are available in the same folders as the respective `Jenkinsfile`.

o-liver reviewed (changes requested):
As there is a lot of changes I think it wise for more people to review it as well.

kbarnold committed:
Add screenshot of example configuration to README.md

Location of the added .png complies with the MkDocs format of our
library, so that we can switch to MkDocs at a later point in time
without much restructuring.

kbarnold commented:
> As there is a lot of changes

I've kept the whitespace changes separate from the content changes. It is easier to see the actual changes to the previous README.md if you go on 'Commits' instead of 'Files Changed' and view the topmost commit there.

1084565 commented on README.md:
```diff
@@ -1,69 +1,110 @@
+application sources. Templates for these configuration files are available in
+the same folders as the respective `Jenkinsfile`.
 
 ## Overview of available pipelines
```
I'd rather see this under `# What you get` that here in `# Configuration` . Any reason why this is here?

kbarnold replied:
Correct. Moved upwards.

1084565 reviewed (changes requested):
Minor changes required:
* Jenkins CI as `Requirement`
*  `## Overview of available pipelines`

The structure follows the compliance guidelines.

1084565 commented on README.md:
```diff
@@ -1,69 +1,110 @@
+# Requirements
 
 * [Installed shared library][piper-library-installation] of project "Piper".
 * Jenkins User with privileges to create jobs.
```
We should name Jenkins 2.60 (?) here.

kbarnold committed:
Move overview of available pipelines section
